force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,33 +147,41 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
+        String scientistString = '';
 
         // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
+        for (String name : scientistNames) {
             // Add the name to the result string, followed by a comma
-
+            if (name.contains('g') || name.contains('G')) {
+                scientistString += name + ',';
+            }
+        }
         // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        if(scientistString.endsWith(',')) {
+            scientistString = scientistString.substring(0, scientistString.length() - 1);
+        }
+        return scientistString;
     }
 
     /**
      * Question 6
-     * This method calculates the absolute value of the difference between of two input numbers that are raised to the 2nd power.
+     * This method calculates the absolute value of the difference between of two input numbers 
+     *   that are raised to the 2nd power.
      * Example: calculateDifferenceOfSquares(3, 4) should return 7.
      * @param a The first number.
      * @param b The second number.
      * @return The absolute value of the difference
      */
     public static Double calculateDifferenceOfSquares(Double a, Double b) {
         // Square the input numbers using the Math.pow() function
+        Double aSquared = a * a;
+        Double bSquared = b * b;
+        Integer result;
 
         // Calculate the absolute difference using Math.abs() function
-
+        result = Integer.valueOf(Math.abs(aSquared - bSquared));
         // Return the result
-        return null; // Replace null with the variable you used to store the result
+        return result; 
```

Comment:
This works but more code than you really need. Can you do this out with ever declaring that result variable ?